### PR TITLE
Eradicated function.bind(this)

### PR DESCRIPTION
**Pull request by Aborysa**
Removed all function.binds in favor of arrow functions.

### CODE REVIEW

File: client/src/components/VotingMenu.js [hunk 1]
```diff
@@ -40,13 +37,13 @@ class VotingMenu extends React.Component {
...
           alternatives={this.props.alternatives}
-          handleChange={this.handleChange}
+          handleChange={(...a) => this.handleChange(...a)}
           selectedVote={this.state.selectedVote}
         />
         <Button
           background
           size="lg"
-          onClick={this.handleClick}
+          onClick={() => this.handleClick(...a)}
```
Comment (duvholt) [inline client/src/components/VotingMenu.js]: Missing `...a`.

File: client/src/components/IssueFormAlternative.js [hunk 2]
```diff
@@ -71,32 +65,32 @@ class IssueFormAlternative extends React.Component {
             <li key={alternative.text}>
               {alternative.text}
               <button
-                onClick={() => removeAlternative(alternative.id)}
+                onClick={(...a) => removeAlternative(alternative.id)}
               >Fjern</button>
-              <button onClick={() => this.openUpdateDialog(alternative.id)}>Endre</button>
+              <button onClick={(...a) => this.openUpdateDialog(alternative.id)}>Endre</button>
```
Comment (duvholt) [inline client/src/components/IssueFormAlternative.js]: Remove `...a` if it isn't used.

File: client/src/components/IssueFormAlternative.js [hunk 3]
```diff
@@ -71,32 +65,32 @@ class IssueFormAlternative extends React.Component {
             <li key={alternative.text}>
               {alternative.text}
               <button
-                onClick={() => removeAlternative(alternative.id)}
+                onClick={(...a) => removeAlternative(alternative.id)}
```
Comment (duvholt) [inline client/src/components/IssueFormAlternative.js]: Remove `...a` if it isn't used.

**Comment by sklirg** [comment]
"In favor of", could you perhaps elaborate on why we should do it one way or another?

**Comment by Aborysa** [comment]
Arrow functions aren't inherently better than using function.bind, but they offer more flexibility than .bind does. Really what I objected to was the use of `this.method = this.method.bind(this)` which in reality translates to `this.method = class.method.bind(this)` or `this.method = this.prototype.method.bind(this)`  in order to "overload" a method and force `this `to refer to the instance when calling `this.property`. Using arrow functions solves this issue since they always keep the callee functions (the method) `this` reference.

**Comment by sklirg** [comment]
Interesting. Looks a bit more maintainable indeed.

@duvholt, care to re-review?

**Pull request merged**